src/ibl_to_nwb/tools/ecephys/spikeglx_to_nwb.py

```diff
@@ -0,0 +1,194 @@
+from typing import Optional
+
+import numpy as np
+from brainbox.io.spikeglx import Streamer
+from hdmf.backends.hdf5 import H5DataIO
+from neuroconv.tools.nwb_helpers import get_module, make_or_load_nwbfile
+from neuroconv.utils import OptionalFilePathType
+from one.api import ONE
+from pynwb import NWBFile
+from pynwb.ecephys import LFP, ElectricalSeries
+
+from src.ibl_to_nwb.tools.ecephys.spikeglxstreamerdatachunkiterator import (
+    SpikeGLXStreamerDataChunkIterator,
+)
+
+IBL_CONFIG = (
+    dict()
+)  # The credentials dict for the IBL database (to be stored in GitHub secret)
+
+
+def add_device(file_streamer: Streamer, nwbfile: NWBFile):
+
+    device_name = file_streamer.pname
+    if device_name in nwbfile.devices:
+        return
+
+    nwbfile.create_device(
+        name=device_name,  # 'probe00'
+        description=f"Version: {file_streamer.version}",
+        # TODO: Where to find the description for this device?
+        manufacturer=file_streamer.meta["typeThis"],  # 'imec'
+    )
+
+
+def add_electrodes(file_streamer: Streamer, nwbfile: NWBFile):
+
+    device_name = file_streamer.pname
+    device = nwbfile.devices[device_name]
+    electrodes_geometry = file_streamer.geometry
+
+    for additional_attribute in [
+        "shank_col",
+        "shank_row",
+        "shank",
+        "flag",
+        "sample_shift",
+        "adc",
+        "ind_in_probe",
+    ]:
+        nwbfile.add_electrode_column(
+            name=additional_attribute,
+            description="no description.",  # TODO: Add descriptions
+        )
+
+    shank_ids = np.unique(electrodes_geometry["shank"]).astype(int)
+    for shank_id in shank_ids:
+        electrode_group = nwbfile.create_electrode_group(
+            name=f"{device_name}_shank{shank_id}",
+            description=f"{device_name}_shank{shank_id}",
+            location="unknown",  # TODO: access from metadata
+            device=device,
+        )
+
+        for electrode_ind in range(len(electrodes_geometry["ind"])):
+            nwbfile.add_electrode(
+                group=electrode_group,
+                filtering="unknown",  # TODO: access from metadata (not sure if available)
+                imp=None,  # TODO: Where to pull impedance from?
+                x=np.nan,
```

Comment:
Several of these fields are not necessary anymore: https://github.com/NeurodataWithoutBorders/pynwb/pull/1448

Reply:
Cool, thanks!